opengever/maintenance/scripts/archival_file_checker.py

```diff
@@ -0,0 +1,394 @@
+        """Return a multiline string representation of the result tables.
+        """
+        all_dossier_stats = self.all_dossier_stats
+        assert all_dossier_stats is not None
+
+        totals = Counter()
+
+        dossier_table = TextTable()
+        dossier_table.add_row((
+            'path',
+            'total_docs_in_dossier',
+            'should_have_archival_file',
+            'with',
+            'without',
+            'missing',
+        ))
+
+        totals['total_resolved_dossiers'] = len(all_dossier_stats)
+
+        for dossier_path, dossier_stats in all_dossier_stats.items():
+            dossier_table.add_row((
+                dossier_path,
+                dossier_stats['total_docs_in_dossier'],
+                dossier_stats['should_have_archival_file'],
+                dossier_stats['with'],
+                dossier_stats['without'],
+                dossier_stats['missing'],
+            ))
+
+            totals['total_docs'] += dossier_stats['total_docs_in_dossier']
+            totals['total_should_have_archival_file'] += dossier_stats['should_have_archival_file']
+            totals['total_missing'] += dossier_stats['missing']
+
+        output = ''
+        output += dossier_table.generate_output()
+        output += '\n\n'
+
+        totals_table = TextTable()
+        totals_table.add_row((
+            'total_resolved_dossiers',
+            'total_docs',
+            'total_should_have_archival_file',
+            'total_missing',
+        ))
+        totals_table.add_row((
+            totals['total_resolved_dossiers'],
+            totals['total_docs'],
+            totals['total_should_have_archival_file'],
+            totals['total_missing'],
+        ))
+
+        output += totals_table.generate_output()
+
+        return output
+
+    def queue_missing(self, missing_by_dossier):
+        """Queue conversion job for documents missing their archival file.
+
+        This method takes a list of dicts, one per dossier that contains at
+        least one document with a missing archival file.
+        """
+        self.log("")
+        self.log("Queueing missing archival files")
+        self.log("=" * 80)
+
+        total_missing = sum([len(g['missing']) for g in missing_by_dossier])
+
+        self.log("Queueing archival file conversion for %s total documents" % total_missing)
+
+        assert IPloneSiteRoot.providedBy(self.context)
+        ann = IAnnotations(self.context)
+        if MISSING_ARCHIVAL_FILE_KEY not in ann:
+            ann[MISSING_ARCHIVAL_FILE_KEY] = IOBTree()
+        queue = ann[MISSING_ARCHIVAL_FILE_KEY]
+
+        for group in missing_by_dossier:
+            dossier = group['dossier']
+            missing = group['missing']
+            print "  Queueing %s documents for %r" % (len(missing), dossier)
+
+            intids = getUtility(IIntIds)
+            dossier_intid = intids.getId(dossier)
+
+            queue[dossier_intid] = IITreeSet()
```

Comment:
So if the queue already existed, we overwrite entries for dossiers that were already in the queue, but would leave entries that are in the queue but not found anymore by this run (for some reason or another). Maybe add a message in the log when `dossier_intid` is already in the queue?

Reply:
Yes, indeed. As suggested, I added logging for the case when we're overwriting/replacing existing queue entries. (Also changed some `print` statements to `self.log` that I seem to have missed 😉 )